docs/branches.md

```diff
@@ -86,9 +87,17 @@ $ # Find the branch we no longer want to track.
 $ # [...]
 # # Actually untrack it.
 $ jj branch untrack <branch name>@<remote name> # Example: jj branch untrack stuff@origin
-$ # From this point on, it won't be imported anymore. 
+$ # From this point on, this remote branch won't be imported anymore.
+$ # The local branch (e.g. stuff) is unaffected. It may or may not still
+$ # be tracking branches on other remotes (e.g. stuff@upstream).
 ```
 
+After the `jj branch untrack`, you can forget the local branch with `jj branch
+forget`, if you wish. (Forgetting is safer than `jj branch delete`, since pushing
```

Comment:
I think I'll postpone this paragraph to a future PR (hopefully forthcoming). I would like to understand the behavior of `jj branch delete` + `jj git push` for untracked branches better.

Reply:
Just fyi, pushing a deleted/existing-but-not-tracking branch to untracked remote is rejected. The underlying model is that push is allowed only if local and remote refs have the same merge base.

Reply:
Thanks, Yuya! This seems to work well, I'll think about how to document it.

Is it also true that `jj branch forget` automatically untracks a branch from all remotes? I was pleasantly surprised that `jj git push --branch qq; jj branch forget qq; jj git fetch` did not recreate the branch `qq`, but I wanted to double-check.

Reply:
> Is it also true that `jj branch forget` automatically untracks a branch from all remotes?

Correct. `jj branch forget` discards all records about the branch, so re-fetched branch is completely new after that.

FWIW, it's undecided whether or not `jj branch forget branch` should delete untracked remote branches. Since untracked remote branches are independent, it might be better to "forget" them separately by `jj branch forget branch@remote`.

Reply:
> FWIW, it's undecided whether or not jj branch forget branch should delete untracked remote branches. Since untracked remote branches are independent, it might be better to "forget" them separately by jj branch forget branch@remote.

I think this makes sense.

Another thing `jj branch forget br` could do is to forget the local branch `br` and do the equivalent of `jj untrack glob:br@*`. This is close to what `jj branch forget; jj fetch` currently does. Perhaps this could be called `jj branch forget --local br`, or perhaps this should be the default behavior and the current one should be an option, I'm unsure.

Reply:
> FWIW, it's undecided whether or not jj branch forget branch should delete untracked remote branches. Since untracked remote branches are independent, it might be better to "forget" them separately by jj branch forget branch@remote.

On second thought, I think the user might want to forget them all at once. So, we should allow `jj branch forget glob:'br@*'`. I'm thinking this could work well with making the `--local` behavior I described the default. We could also have `jj branch forget --with-tracked-remotes br`.